ResidentInformationApi/V1/Gateways/HousingInformationGateway.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using ResidentInformationApi.V1.Boundary.Requests;
+using ResidentInformationApi.V1.Domain;
+using ResidentInformationApi.V1.Gateways.Helpers;
+
+namespace ResidentInformationApi.V1.Gateways
+{
+    public class HousingInformationGateway : IHousingInformationGateway
+    {
+        private readonly HttpClient _client;
+        private readonly string _baseUrl;
+
+        public HousingInformationGateway(HttpClient client)
+        {
+            _client = client;
+            _baseUrl = Environment.GetEnvironmentVariable("HOUSING_API_ENDPOINT");
+        }
+
+        public async Task<List<HousingResidentInformation>> GetResidentInformation(ResidentQueryParam rqp)
+        {
+            var queryDictionary = DictionaryBuilder.BuildQueryDictionary(rqp);
+            var rqpString = new FormUrlEncodedContent(queryDictionary).ReadAsStringAsync().Result;
```

Comment:
It looks good, to slim it down even further you could add this line to the helper method as well and have it as a query string builder. I can't think of any cases where you will need the dictionary instead of the string.

Reply:
Good shout! I've made that change